scripts/setup_linux.sh

```diff
@@ -114,98 +128,91 @@ cd "${GARAGE_DIR}"
+git clone https://github.com/glfw/glfw.git "${GLFW_DIR}"
+cd "${GLFW_DIR}"
 git checkout 0be4f3f75aebd9d24583ee86590a38e741db0904
 mkdir glfw-build
 cd glfw-build
 cmake -DBUILD_SHARED_LIBS=ON -DGLFW_BUILD_EXAMPLES=OFF \
   -DGLFW_BUILD_TESTS=OFF -DGLFW_BUILD_DOCS=OFF ..
 make -j"$(nproc)"
 make install
-cd ../../
-rm -rf glfw
+cd "${GARAGE_DIR}"
+
+# Leave a note in ~/.bashrc for the added environment variables
+if [[ "${_arg_set_envvar}" = on ]]; then
+  echo -e "\n# Added by the garage installer" >> "${BASH_RC}"
+fi
 
 # Set up MuJoCo
-if [[ ! -f "${HOME}"/.mujoco/ ]]; then
+if [[ ! -d "${HOME}"/.mujoco/mjpro150 ]]; then
   mkdir "${HOME}"/.mujoco
-  wget https://www.roboti.us/download/mjpro150_linux.zip -O mujoco.zip
-  unzip mujoco.zip -d "${HOME}"/.mujoco
-  rm mujoco.zip
+  MUJOCO_ZIP="$(mktemp --suffix=_mujoco.zip)"
+  wget https://www.roboti.us/download/mjpro150_linux.zip -O "${MUJOCO_ZIP}"
+  unzip -u "${MUJOCO_ZIP}" -d "${HOME}"/.mujoco
 else
-  printf '\e[0;31m%s\e[0m\n' "Mujoco is already installed!"
+  print_warning "MuJoCo is already installed"
 fi
 # Configure MuJoCo as a shared library
 export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150/bin"
-LD_LIB_ENV_VAR="LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:${HOME}/.mujoco"
-LD_LIB_ENV_VAR="${LD_LIB_ENV_VAR}/mjpro150/bin\""
+LD_LIB_ENV_VAR="LD_LIBRARY_PATH=\"\$LD_LIBRARY_PATH:${HOME}/.mujoco/mjpro150"
+LD_LIB_ENV_VAR="${LD_LIB_ENV_VAR}/bin\""
 if [[ "${_arg_set_envvar}" = on ]]; then
-  printf "\\n# Added by the garage installer\\n" >> "${BASH_RC}"
-  printf "export ${LD_LIB_ENV_VAR}\\n" >> "${BASH_RC}"
+  echo "export ${LD_LIB_ENV_VAR}" >> "${BASH_RC}"
 fi
 
 # Set up conda
 hash conda 2>/dev/null || {
+  CONDA_INSTALLER="$(mktemp --suffix=_miniconda.sh)"
+  sudo chmod u+x "${CONDA_INSTALLER}"
   wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh \
-    -O miniconda.sh
-  sudo bash miniconda.sh -b -p /opt/conda
-  rm miniconda.sh
-  sudo ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh
+    -O "${CONDA_INSTALLER}"
+  sudo bash "${CONDA_INSTALLER}" -b
+  # Add conda to executable programs
+  CONDA_PATH="${HOME}/miniconda2/bin/"
+  export PATH="$PATH:${CONDA_PATH}"
+  PATH_ENV_VAR="PATH=\"\$PATH:${CONDA_PATH}\""
+  if [[ "${_arg_set_envvar}" = on ]]; then
+    echo "export ${PATH_ENV_VAR}" >> "${BASH_RC}"
+  fi
 }
-# Add conda to executable programs
-export PATH="$PATH:/opt/conda/bin"
-PATH_ENV_VAR="PATH=\"\$PATH:/opt/conda/bin\""
-if [[ "${_arg_set_envvar}" = on ]]; then
-  printf "\\n# Added by the garage installer\\n" >> "${BASH_RC}"
-  printf "export ${PATH_ENV_VAR}\\n" >> "${BASH_RC}"
-fi
 conda update -q -y conda
 
 # Create conda environment
 conda env create -f environment.yml
 if [[ "${?}" -ne 0 ]]; then
-  printf '\e[0;31m%s\e[0m\n' "Error: conda environment could not be created"
-  exit 1
+  print_error "Error: conda environment could not be created"
 fi
-sudo rm -rf /opt/conda/pkgs/*
 
 # Extras
 source activate garage
```

Comment:
Specify a tensorflow version. Right now 1.9 works but 1.10 has some warnings we need to address